internal/compose/compose.go

```diff
@@ -141,8 +143,9 @@ func NewProject(name string, paths ...string) (*Project, error) {
 	}
 
 	c := Project{
-		name,
-		paths,
+		name:             name,
+		dcMajorVersion:   "",
```

Comment:
As suggested in the other comment, `dcMajorVersion` can be replaced with `dockerComposeV1` (bool).

Then in the `NewProject`, you can evaluate its value. 

BTW I hope that the `Project` isn't created somewhere just with `{}`.